sympy/geometry/point.py

```diff
@@ -80,49 +82,225 @@ class Point(GeometryEntity):
     def __new__(cls, *args, **kwargs):
         evaluate = kwargs.get('evaluate', global_evaluate[0])
 
-        if iterable(args[0]):
-            if isinstance(args[0], Point) and not evaluate:
-                return args[0]
-            args = args[0]
-
-        # unpack the arguments into a friendly Tuple
-        # if we were already a Point, we're doing an excess
-        # iteration, but we'll worry about efficiency later
-        coords = Tuple(*args)
-        if any(a.is_number and im(a) for a in coords):
-            raise ValueError('Imaginary coordinates not permitted.')
+        # unpack into coords
+        if len(args) == 1:
+            coords = args[0]
+        else:
+            coords = args
+
+        # check args and handle quickly handle Point instances
+        if not is_sequence(coords) and not isinstance(coords, Point):
+            raise TypeError(filldedent('''
+                Expecting sequence of coordinates, not `{}`'''
+                .format(func_name(coords))))
+        dimension = as_int(kwargs.get('dim', len(coords)))
+        if isinstance(coords, Point):
+            if dimension == len(coords):
+                return coords
+            evaluate = False
+            coords = Tuple(*coords)
+        else:
+            coords = Tuple(*coords)
+            # check only coords that weren't in a Point already
+            if any(a.is_number and im(a) for a in coords):
+                raise ValueError('Imaginary coordinates not permitted.')
+            if not all(isinstance(a, Expr) for a in coords):
+                raise TypeError('Coordinates must be valid SymPy expressions.')
+
+        # adjust size
+        if len(coords) > dimension:
```

Comment:
It's only bumped down if the user says they want it so by supplying the `dim` keyword. We could add a `warn` keyword to allow a warning when dimensions are being reduced.